File: java/src/test/java/problems1_100/SortColorsTest.java
package problems1_100;

import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class SortColorsTest extends SampledTest {
    @Test
    public void test() {
        for (SortColors solution : SortColors.values()) {
            for (SortColorsSample sample : getSamples(SortColorsSample[].class)) {
                int[] nums = sample.input.nums;
                solution.sortColors(nums);
                assertWithMessage(sample.getMessage())
                    .that(nums)
                    .asList()
                    .containsExactlyElementsIn(
                        Arrays
                            .stream(sample.output)
                            .boxed()
                            .toList()
                    );
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/SortColorsTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class SortColorsTest : SampledTest() {
    @Test
    fun test() =
        SortColors.entries.forEach { solution ->
            getSamples<Array<SortColorsSample>>().forEach { (input, output, message) ->
                val nums = input.nums
                solution.sortColors(nums)
                assertWithMessage(message)
                    .that(nums)
                    .asList()
                    .containsExactlyElementsIn(output.toList())
                    .inOrder()
            }
        }
}
